contracts/StablecoinConverter.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+
+import "./EpochTokenLocker.sol";
+import "./libraries/IdToAddressBiMap.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "./libraries/IdToAddressBiMap.sol";
+
+
+contract StablecoinConverter is EpochTokenLocker {
+    using SafeMath for uint;
+    using BytesLib for bytes32;
+
+    event OrderPlacement(
+        address owner,
+        uint16 buyToken,
+        uint16 sellToken,
+        bool sellOrderFlag,
+        uint32 validFrom,
+        uint32 validUntil,
+        uint256 buyAmount,
+        uint256 sellAmount
+    );
+
+    event OrderCancelation(
+        address owner,
+        uint id
+    );
+
+    //outstanding volume of an order is encoded as buyAmount or sellAmount depending on sellOrderFlag
+    struct Order {
+        uint16 buyToken;
+        uint16 sellToken;
+        uint32 validFrom;  // order is valid from validFrom inclusively
+        uint32 validUntil;  // order is valid till validUntil inclusively
+        bool sellOrderFlag;
+        uint128 buyAmount;
+        uint128 sellAmount;
+    }
+
+    // User-> Order
+    mapping(address => Order[]) public orders;
+
+    IdToAddressBiMap.Data private registeredTokens;
+    uint public MAX_TOKENS;
+    uint16 public numTokens = 0;
+
+    constructor(uint maxTokens) public {
+        MAX_TOKENS = maxTokens;
+    }
+
+    function addToken(address _tokenAddress) public {
+        require(numTokens < MAX_TOKENS, "Max tokens reached");
+        require(
+            IdToAddressBiMap.insert(registeredTokens, numTokens, _tokenAddress),
+            "Token already registered"
+        );
+        numTokens++;
```

Comment:
Does it make sense to include the count in the data of IdToAddressBiMap?
Also, wouldn't it make sense to provide a way to remove a token too?

Reply:
Yeah, let's push this tokenHandling into another PR.